crates/nu-cli/src/completions/arg_value_dynamic_completion.rs

```diff
@@ -0,0 +1,57 @@
+use crate::completions::{
+    Completer, CompletionOptions, SemanticSuggestion, completion_options::NuMatcher,
+};
+use nu_protocol::{
+    DeclId, Span,
+    engine::{ArgType, Stack, StateWorkingSet},
+};
+use reedline::Suggestion;
+
+pub struct ArgValueDynamicCompletion<'a> {
+    pub decl_id: DeclId,
+    pub arg_type: ArgType<'a>,
+}
+
+impl<'a> Completer for ArgValueDynamicCompletion<'a> {
+    fn fetch(
+        &mut self,
+        working_set: &StateWorkingSet,
+        stack: &Stack,
+        prefix: impl AsRef<str>,
+        span: Span,
+        offset: usize,
+        options: &CompletionOptions,
+    ) -> Vec<SemanticSuggestion> {
+        // the `prefix` is the value of a flag
+        // if user input `--foo abc`, then the `prefix` here is abc.
+        // the name of flag is saved in `self.flag_name`.
+        let mut matcher = NuMatcher::new(prefix, options);
+        let mut add_suggestion = |value: String| {
```

Comment:
nitpick: What's this closure for? If it's just to decrease indentation level, you can do that by using `let Some(items) = decl... else { return vec![] };` below

Reply:
Probably copied from FlagCompletion where I forgot to clean 😅. I think rustc will inline it automatically. If that's the case, it's actually more readable IMHO.